### DISCUSSION TITLE
请问 vc-drawings 组件的预添加数据如何设置自定义属性

### DISCUSSION THREAD

**Discussion by Qvod** [zouyaoji/vue-cesium#435] [Q&A] [answered]
![截屏2023-08-24 18 24 47](https://github.com/zouyaoji/vue-cesium/assets/5636321/93621445-96d7-44c2-8058-65450d32e8a8)

请问如何给preRenderDatas添加自定义属性，比如ID。我想实现预加载点位数据，编辑后可以通过自定义的ID来修改远程数据。

**Comment by zouyaoji** [comment, marked as answer]
你好，现在更推荐给组件绑定ref，通过ref获取到给每个action的renderDatas来预添加数据。例如
```
drawingsRef.value.getDrawingActionInstances()[2].cmpRef.value.renderDatas.value
```
自定义属性也可以一并加进去。
